arch/RISCV/riscv_decode.gen.inc

```diff
@@ -0,0 +1,7056 @@
+#include <stdint.h>
+
+#include <stddef.h>
+
+#include <string.h>
+#include "riscv_ast.gen.inc"
+
+
+void decode(struct ast *tree, uint64_t binary_stream) {
+    uint64_t op = 0xFFFFFFFFFFFFFFFF ;
+    switch (binary_stream & 0x000000000000007F) {
+      case 0x37: 
+            op = RISCV_LUI ;
+      break; 
+      case 0x17: 
+            op = RISCV_AUIPC ;
+      break; 
+ 
+    }
+    if (op != 0xFFFFFFFFFFFFFFFF) { 
+        uint64_t rd = (binary_stream & 0x0000000000000F80)>>7 ;
+        uint64_t imm = (binary_stream & 0x00000000FFFFF000)>>12 ;
+        tree->ast_node_type = RISCV_UTYPE ;
+        tree->ast_node.utype.imm = imm;
+        tree->ast_node.utype.rd = rd;
+        tree->ast_node.utype.op = op;
+        return ;
+    }
+  if ((binary_stream & 0x000000000000007F == 0x6F)) { 
+      uint64_t rd = (binary_stream & 0x0000000000000F80)>>7 ;
+      uint64_t imm_7_0 = (binary_stream & 0x00000000000FF000)>>12 ;
+      uint64_t imm_8 = (binary_stream & 0x0000000000100000)>>20 ;
+      uint64_t imm_12_9 = (binary_stream & 0x0000000001E00000)>>21 ;
+      uint64_t imm_18_13 = (binary_stream & 0x000000007E000000)>>25 ;
+      uint64_t imm_19 = (binary_stream & 0x0000000080000000)>>31 ;
+      tree->ast_node_type = RISCV_JAL ;
+      tree->ast_node.riscv_jal.imm = (imm_19 << 23) | (imm_7_0 << 15) | (imm_8 << 14) | (imm_18_13 << 8) | (imm_12_9 << 4) | (0x0 << 0);
+      tree->ast_node.riscv_jal.rd = rd;
+      return ;
+  }
+  if ((binary_stream & 0x000000000000007F == 0x67) && ((binary_stream & 0x0000000000007000)>>12 == 0x0)) { 
```

Comment:
The `binary_stream & 0x000000000000007F == 0x67` also in a helper please. E.g.
`inline bool test_bits64(uint64_t bytes, size_t start, size_t n, uint64_t expected)`